shared/src/telemetry/posthog.rs

```diff
@@ -38,109 +37,94 @@ impl Config {
-    if let Some(config) = Config::from_env() {
-        if config.telemetry.as_deref() == Some("true") {
-            let uuid_dir = "/var/lib/postgresql/data";
-            let extension_name;
-            let file_content = if Path::new(uuid_dir)
-                .join(format!("{}_uuid", PARADEDB_NAME.to_lowercase()))
-                .exists()
-            {
-                extension_name = PARADEDB_NAME;
-                fs::read_to_string(
-                    Path::new(uuid_dir).join(format!("{}_uuid", PARADEDB_NAME.to_lowercase())),
-                )
-            } else {
-                extension_name = PG_BM25_NAME;
-                fs::read_to_string(Path::new(uuid_dir).join(format!("{}_uuid", PG_BM25_NAME)))
-            };
-
-            let distinct_id = match file_content {
-                Ok(uuid) => uuid,
-                Err(_) => {
-                    warn!("telemetry enabled but uuid file is empty!");
-                    return;
-                }
-            };
-
-            let endpoint = format!("{}/capture", config.posthog_host);
-            let data = json!({
-                "api_key": config.posthog_api_key,
-                "event": format!("{} Connection Started", extension_name),
-                "distinct_id": distinct_id,
-                "properties": {
-                    "commit_sha": config.commit_sha
-                }
-            });
-
-            let client = reqwest::blocking::Client::new();
-            let response = client
-                .post(endpoint)
-                .header("Content-Type", "application/json")
-                .body(data.to_string())
-                .send();
-
-            if let Err(e) = response {
-                warn!("Error sending telemetry request: {}", e);
-            }
+fn generate_and_save_uuid(uuid_file: &str) -> String {
+    let new_uuid = uuid::Uuid::new_v4().to_string();
+    fs::write(uuid_file, &new_uuid).expect("Unable to write UUID to file");
+    new_uuid
+}
+
+fn send_telemetry_data(settings: &TelemetrySettings, extension_name: String) {
+    let telemetry_data = read_telemetry_data(extension_name.clone());
+    pgrx::log!("telemetry data: {:?}", telemetry_data);
+
+    let endpoint = format!("{}/capture", settings.posthog_host);
+    let data = json!({
+        "api_key": settings.posthog_api_key,
+        "event": format!("{} Deployment", extension_name),
+        "distinct_id": settings.distinct_id,
+        "properties": {
+            "commit_sha": settings.commit_sha,
+            "telemetry_data": telemetry_data
         }
+    });
+
+    let client = reqwest::blocking::Client::new();
+    let response = client
+        .post(endpoint)
+        .header("Content-Type", "application/json")
+        .body(data.to_string())
+        .send();
+
+    if let Err(e) = response {
+        eprintln!("Error sending request: {}", e);
+    }
+}
+
+// Adjusted function to use the new structure
+pub fn read_and_send_telemetry_data(extension_name: String) {
```

Comment:
This is an unnecessary abstraction. Just call the `initialize_telemetry` and `send_telemetry_data` functions when you need to call them.